### ISSUE TITLE
TODO list for neural networks

### ISSUE THREAD

**Issue by rohskopf** [FitSNAP/FitSNAP#116]
- Add a feature to calculate validation and/or test errors on the fly, like we do with training error.
- Add a function or post-processing script to print the LAMMPS-ready PyTorch models, which was previously done with features in the MLIAP python package. Here we avoid importing MLIAP because of the https://github.com/lammps/lammps/issues/3204.
- Add NN fitting for multiple element types by creating a separate NN for each atom type.
- Ignore the large descriptor gradient array when fitting only to energies.
- Get working on GPU.
-  Fix the training/validation group feature for NNs. Some users may want to specify which groups are used for training and validation. If these options are used, then don't randomly splitting the entire dataset into training/validation sets.
- Group weights for different configs. If force weights for a group are zero, don't bother calculating forces for that group. This may prove challenging since configs might be randomly shuffled before training, we need to carry the specified weight with the config during shuffling.

**Comment by rohskopf** [comment]
These tasks are complete!

> Add a feature to calculate validation and/or test errors on the fly, like we do with training error.
https://github.com/FitSNAP/FitSNAP/pull/118

> Add a function or post-processing script to print the LAMMPS-ready PyTorch models
https://github.com/FitSNAP/FitSNAP/pull/130
https://github.com/FitSNAP/FitSNAP/pull/131

> Add NN fitting for multiple element types by creating a separate NN for each atom type.
https://github.com/FitSNAP/FitSNAP/pull/130

> Get working on GPU.
https://github.com/FitSNAP/FitSNAP/pull/119

> Train/validation groups and weights
https://github.com/FitSNAP/FitSNAP/pull/132
